EPIC: Compare View Delta, UX Refactor, Profile Data Details and Interface Unification

cjdoherty opened the pull request:
[EPIC Feature] Comparison View for all uploaded evaluations.

#### MVP 1.0 
- [x] upload doesn't seem to load the same way in the delta view - the failed tests by severity are not the same in the results view vs the compare view
- [x] Change the word "upload" to "load"
- [x] After deleting all loaded results, blank the screen and bring back the default load screen - see current HL deployment
- [x] In compare view - make the control id column expand all in the row
- [x] Bring back the `clear filters` button :)
- [x] ensure that overlay profile compliance dial results count correctly under the comparison view. For example, the triple_overlay_profile_example.json counts triple the number of tests. (600 instead of only 200) @cjdoherty 
- [x] for profile information add back in the profile dependency tree (except feel free to make it vertical to save space) see screen shot and samples below from heimdall lite 1.0. Note, overlays branching to dependent baselines/overlays. @cjdoherty 
- [x] add functional and unit tests @samuelsunvold @cjdoherty 
- [x] remove unneed comments from the final code @cjdoherty @rbclark 
- [ ] get with @rbclark on the proper use of `console.log` and `try` / `catch` on error logging @cjdoherty @rbclark 
- [x] profile details only activate with four or more results is loaded
- [x] if you load 4 results of two different types the show only changed breaks
  - example Nginx good and bad with cis foundations 
- [x] Add checkbox multiple loading of files on the Upload Nexus so we can easily and quickly show the Compare feature
- [x] Do we want to keep the interface consistent and have an 'expand all' and 'close all' on the compare rows as we do in the other interface elements? @ejaronne @cjdoherty 
- [x] The `export as json` tool seems to be broken or not wired up so we should wire it @robthew 
- [x] The top component that shows loaded file(s) info is not loading data from the vuex store, please update it to actually show the loaded file or files data @cjdoherty @robthew 
- [x] Fix severity button in the delta table compare, it is being colored and makes the text area of the severity be the same color as the status button - bleed over from the status button? @cjdoherty
- [x] The save disk icon is still showing in the side menu in heimdall-lite mode - missing v-if referencing the vuex state for server vs lite mode? @cjdoherty  
- [x] fix viewport issues with column size and text going vertical and text in buttons not reacting correctly
- [x] add an icon that lets users know they can collapse the top section - https://materialdesignicons.com/icon/arrow-collapse
- [x] add example sets for data show and tell to include: 
- [x] Update the 'samples' data to use the vuejs 'assets' area rather than 'master branch' 

  **Note:** you must commit files to the master branch in the samples dir for now until we fix this to use the `assets` directory correctly for this application data.

  - [ ] add at least one Sponsor overlay so we can show the controls going from enabled to Not Applicable @ejaronne 
  - [x] Nginx good / bad ( we need to update bad-nginx to not have the extra data about rony's account ) @aaronlippold 
  - [x] RedHat good / bad @ejaronne @HackerShark 
  - [ ] AWS Foundations good / bad @rx294 
- [x] Make show changed controls the default starting state in the display
- [x] Allow the user to collapse the graphs to enable review of data
- [x] add the flat class to the graphs so they don’t look like they’re floating on top of that they’re integrated into the background
- [x] Make sure we test the mobile, tablet and desktop views in both horizontal and vertical orientation
- [x] add smart expand to know when only a single click ( 1 or 2 loaded results ) or multi-click ( 3 or more is needed
- [x] Make the three tabs of our data display standard even when we only have 1 or 2 files loaded
- [x] If I am in the results view and I load a new file, it is not loading that file into the compare view automatically and I have to go back to the hamburger menu and select it etc. which isn't a great user experience.

#### Next iteration 

 - [ ] add the ability to create a 'group' or 'set' for the results so we can do a 'delta' of a system which may have a few results

Fixes #41

cjdoherty committed:
WIP Comparison view

Mitriol committed:
Type name clarifications

Signed-off-by: Jacob Henry <jchenry@mitre.org>

Mitriol committed:
Fixed delta_sets()

Signed-off-by: Jacob Henry <jchenry@mitre.org>

Mitriol committed:
Sourced from fix

Signed-off-by: Jacob Henry <jchenry@mitre.org>

cjdoherty committed:
Bug fixes

cjdoherty committed:
merged

cjdoherty committed:
fixed bug

cjdoherty committed:
commented out log in popup causing bug

cjdoherty committed:
Fixed ControlRowCol formatting issues

cjdoherty committed:
fixed coloring

cjdoherty committed:
added data charts

Mitriol committed:
Fixed apexcharts in tabs

Signed-off-by: Jacob Henry <jchenry@mitre.org>

cjdoherty committed:
file selection WIP

cjdoherty committed:
WIP file select

Mitriol committed:
Fixed some possible cyclic reference issues

Mitriol committed:
Merge branch 'delta-view' of https://github.com/mitre/heimdall-lite into delta-view

cjdoherty committed:
multi file support

Mitriol committed:
Added better popups for no files / none selected / all filtered

Signed-off-by: Jacob Henry <jchenry@mitre.org>

Mitriol committed:
Made different views explicit in sidebar

Signed-off-by: Jacob Henry <jchenry@mitre.org>

Mitriol committed:
Fixed up sidebar persistence, as well as miscellaneous code cleanup and bugfixing

Signed-off-by: Jacob Henry <jchenry@mitre.org>

aaronlippold commented:
Can we also add bad-nginx to the samples and a couple other 'delta's' we can test

aaronlippold commented:
This also seems to be an issue in the viewport. So we may need to adjust the colun span in the flex and some of the same issues we had to resolve with the datatable layout.

![image](https://user-images.githubusercontent.com/1486440/84089658-87c57800-a9bd-11ea-90ee-dc51d34e38ae.png)

aaronlippold commented:
I added the stuff to the PR description to help us keep track

cjdoherty committed:
auto compares with one click with 2 files

cjdoherty committed:
- updates files getter in compare
- added getter for checking if evaluations are of same profile

cjdoherty committed:
changed "all reports" button

cjdoherty committed:
- added resizing for compare.vue
- limited number of visible tests depending on screen size
- added arrows to shift through tests in compare.vue

cjdoherty committed:
-fixed bug by making selected buttons clear when new
file is added
- added index to profile names to keep track
- added slide group for status charts

cjdoherty committed:
cleanup

ejaronne commented:
For compare, show all results even if the test numbers don't align. Emulate how it currently works on Heimdall server, please. 

When comparing more than 2 result sets, show status "Status by Results file" first and on the left.

Change the word "Evaluation" to "Results" everywhere.

Thanks!

cjdoherty committed:
auto selects newly uploaded files in compare view

cjdoherty committed:
tabs now show in compare with 1 or 2 files

cjdoherty committed:
changed from "evaluation" to "results"

cjdoherty committed:
same as above ^^

cjdoherty committed:
-fixed functionality to be similar to old heimdall
when comparing results of differing profiles.
- fixed bug with upper bound of compare severity chart

ejaronne commented:
- Change "Test Evaluations" to "Test Results"
- remove "No Data" - just leave blank
- Change "Only Changed Tests" to read "Display Only Changed Results"
- Left to right, list "Status by Results File", % Compliance, Failed Tests by Severity"
- Default to "Status by Results File" always on first load of a set comparison
- Label first column "Test ID"
- Allow ability to sort by Test ID and other columns
- allow Search just like a results page

cjdoherty committed:
All but bottom 3 bullets above

ejaronne commented:
- Above each column, display: column number, then filename under that, then date/time stamp under that
- Also, move "Display Only Changed Results" above the columns and to right of "Test Results":
![image](https://user-images.githubusercontent.com/34140975/84542725-0d03a200-acc8-11ea-9fd2-cc1ef5f6a170.png)

ejaronne commented:
- Ensure pop-ups can be read in light or dark mode
- include number and filename in header of the pop-up

![image](https://user-images.githubusercontent.com/34140975/84542830-3b817d00-acc8-11ea-9d4e-9f47d492beaa.png)

cjdoherty committed:
Added "Test ID label and start_times to each row

cjdoherty committed:
moved "Display Only Changed Results" checkbox

cjdoherty committed:
fixed bug with start time

cjdoherty committed:
Added test ID sort

cjdoherty committed:
- fixed translucent text bug
- added searchbar (not functiontioning yet

cjdoherty committed:
Added search functionality

cjdoherty committed:
- moved checkbox and set it to start checked

cjdoherty committed:
added file times to status charts

cjdoherty committed:
-sizing adjustments
- adjusted tabs so the slide item does not look ugly
- added collapse feature for tabs WIP

cjdoherty committed:
covered by commit above ^^

aaronlippold commented:
There are also merge conflicts on the PR.

cjdoherty committed:
Merge branch 'master' into delta-view

cjdoherty committed:
- lint formatting and npm errors

cjdoherty committed:
- Added collapse and fixed sizing

cjdoherty committed:
changed sizing

cjdoherty committed:
fixed bug with search

cjdoherty committed:
edited graph so there are a proper number of ticks

aaronlippold committed:
- added bad_nginx to the samples tab
- added icons and boolean state to the state of the graphs tab
- updated the Samples titles
- cleaned Compare component of commented code

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold reviewed (changes requested):
There are a few small things that I added to the checkboxes and tried to document. I also closed out a few of the things I added. Once we - I think dear lord - fix the things now on the list we should be good to go.

aaronlippold committed:
- updated the icons in the tools sidebar to be a bit more clear

Signed-off-by: Aaron Lippold <lippold@gmail.com>

cjdoherty committed:
Evaluation info component displays the selected info

cjdoherty committed:
added collapse icon to tab in compare

cjdoherty committed:
removed sidebar save button when not in server.

cjdoherty committed:
Fixed Search clear and chart start times

cjdoherty committed:
Moved samples to assets

cjdoherty committed:
Merge branch 'master' of https://github.com/mitre/heimdall-lite into delta-view

cjdoherty committed:
fixed export json /  added samples

cjdoherty committed:
Added carousel evaluation info for multiple files

cjdoherty committed:
Added slide group for evaluation info

cjdoherty committed:
cleaned up slide group for eval info

rbclark commented on src/views/Results.vue:
```diff
@@ -380,16 +394,22 @@ export default class Results extends ResultsProps {
-    else if (ids.length > 1) {
-      this.$router.push(`/results/all`);
+  get info_width(): number {
+    let mod = getModule(ServerModule, this.$store);
+    if (mod.serverMode == undefined) {
+      mod.server_mode();
+    }
```
There has to be some way to load whether or not the application is in server mode on startup. Perhaps using on app `mounted` or something so that this check doesn't have to be done every time, this PR alone has 5 instances of this block of code in it.

rbclark replied:
Nevermind, this is refactored in #277, I'll have to resolve this in my PR when this is merged.

rbclark commented on src/components/cards/EvaluationInfo.vue:
```diff
@@ -139,39 +142,68 @@ export default class EvaluationInfo extends EvaluationInfoProps {
+      mod.server_mode();
+    }
+    if (mod.serverMode) {
+      return 500;
```
Some sort of comment explaining why these numbers were chosen would be nice, without reading the function name at first glance I thought this was returning an HTTP error code. At least a comment explaining the reason for each of these numbers will make it easier for future developers.

rbclark commented on src/components/cards/EvaluationInfo.vue:
```diff
@@ -139,39 +142,68 @@ export default class EvaluationInfo extends EvaluationInfoProps {
 
   watch() {
-    console.log("Prop changed: " + this.filter);
+    console.log("Prop changed: " + this.file_filter);
```
The devtools does this for you instead of requiring use of console.log: https://vuejs.org/v2/cookbook/debugging-in-vscode.html#Vue-Devtools

cjdoherty committed:
added checkbox selection for samples

ejaronne commented:
"InSpec Version" is the incorrect label when displaying output from Heimdall_tools. Heimdall_tools sets the platform.name to "Heimdall Tools" and the platform.release and version to that of Heimdall_tools. So what is a more general term for "InSpec Version"? "Tool Version"? I'm not sure Heimdall_tools can always identify the target platform that was scanned by the orginal tool.

cjdoherty committed:
fixed sample selection bug

cjdoherty committed:
fixed wierd scroll in upload modal

cjdoherty committed:
fixed scroll bug in in upload modal

cjdoherty committed:
combined profile data with evaluation info component

cjdoherty committed:
code cleanup and commenting

cjdoherty committed:
fixed bug where prof data only worked with > 3 files

cjdoherty committed:
fix for "Only Changed Results", not optimized

rbclark reviewed (changes requested):
Please see inline comments from before.

rbclark commented on src/views/Results.vue:
```diff
@@ -240,9 +269,9 @@ export default class Results extends ResultsProps {
    * no file uploaded
    */
-  mounted() {
+  /*mounted() {
```
When removing code if it is no longer needed it should just be deleted, not commented out. If it is ever needed in the future again the git history can be used to find it.

Bialogs reviewed (dismissed):
This needs to be rebased.

ejaronne commented:
For profile dependency tree displays:
### overlay profile (see triple_overlay_profile_example.json)
![image](https://user-images.githubusercontent.com/34140975/86652271-4327ff00-bfb2-11ea-8591-cbb91138221d.png)

### wrapper profile, e.g., https://github.com/chris-rock/acme-inspec-profile.git 
![image](https://user-images.githubusercontent.com/34140975/86652291-46bb8600-bfb2-11ea-8334-d2fcadf7ec49.png)

cjdoherty committed:
resolved merge conflicts

cjdoherty committed:
updated gitignore

cjdoherty committed:
fixed data counting problem

cjdoherty committed:
fixed data counting in severity chart

cjdoherty committed:
fixed profile heirarchy of prof data

cjdoherty committed:
added tree to view overlayed profiles to profdata

cjdoherty committed:
cleanup and comment

cjdoherty committed:
added triple overlay to samples

ejaronne commented:
Delta view does not show overlay content in the delta view the way the results view does:

Using the triple_overlay_example:

### This is the results view:
![image](https://user-images.githubusercontent.com/34140975/87374004-95d85c80-c558-11ea-8bc0-604a5984e625.png)

### This is the delta view:
![image](https://user-images.githubusercontent.com/34140975/87374357-b9030c00-c558-11ea-9795-5f35aeaf68ed.png)

cjdoherty committed:
added "change data" and "all data" options
- displays tabs for all data
- displays tests and changed attributes for changed data

cjdoherty committed:
changed upload to load

cjdoherty committed:
merge conflicts

cjdoherty committed:
linting

cjdoherty committed:
added clear button

cjdoherty committed:
goes to landing page is no files are loaded

rbclark commented on src/views/Results.vue:
```diff
@@ -124,45 +156,59 @@
       style="margin-top: 44px;"
       v-model="filter_snackbar"
-      :timeout="10000"
+      :timeout="10000000"
```
I assume the point of this is to have the warning hang around the whole time, in which case the way to do this is:

> Use -1 to keep open indefinitely (0 in version < 2.3 )

cjdoherty committed:
made header changes only show with change

rbclark commented on src/components/cards/comparison/ChangeItem.vue:
```diff
@@ -1,28 +1,61 @@
+      sm="3"
+      md="3"
+      lg="3"
+      xl="3"
```
This shouldn't be necessary, if a value isn't set for a breakpoint then it just inherits from the one before it. The following is equivalent:

```ruby
    <v-col
      class="pa-0"
      v-for="(value, i) in values"
      :key="i"
      cols="4"
      sm="3"
>
```

rbclark commented on src/components/cards/comparison/ChangeItem.vue:
```diff
@@ -1,28 +1,61 @@
-    </v-col>
-    <v-col class="pa-1" cols="1" align-self="center">
-      <v-icon> arrow_forward </v-icon>
+    <v-col cols="3" xs="3" sm="2" md="2" lg="1" xl="1" class="pa-0">
```
If the breakpoint for `md` is the same as the one for `sm` then it can be omitted, same for `xl` since it is the same as lg. Also if `xs` value is the same as the number of cols then it can also be omitted (all the way up until a breakpoint that you actually specify a value for).

rbclark commented on src/components/cards/comparison/CompareRow.vue:
```diff
@@ -1,38 +1,87 @@
-        <v-card class="fill-height" color="info">
-          <v-card-title> {{ hdf_controls[0].wraps.id }} </v-card-title>
-        </v-card>
+      <v-col cols="3" xs="3" sm="2" md="2" lg="1" xl="1" class="pt-0">
```
Same as previous comment involving cols.

rbclark commented on src/components/cards/comparison/CompareRow.vue:
```diff
@@ -1,38 +1,87 @@
+        sm="3"
+        md="3"
+        lg="3"
+        xl="3"
```
Same as previous comment involving cols.

rbclark commented on src/components/cards/comparison/CompareRow.vue:
```diff
@@ -1,38 +1,87 @@
       <!-- Depending on selection, more details -->
       <!-- <transition-group> -->
-      <v-col cols="12" v-if="delta" key="delta">
+      <!--v-col cols="12" v-if="delta" key="delta">
```
Do not leave commented code.

rbclark commented on src/components/cards/comparison/CompareRow.vue:
```diff
@@ -1,38 +1,87 @@
+        sm="3"
+        md="3"
+        lg="3"
+        xl="3"
```
See previous comment on cols.

rbclark commented on src/components/cards/comparison/CompareRow.vue:
```diff
@@ -42,39 +91,77 @@ import {context} from 'inspecjs';
+  //     }
+  //   }
+  //   return det_ind;
+  // }
```
Do not leave commented out code.

rbclark commented on src/components/cards/comparison/DeltaView.vue:
```diff
@@ -2,53 +2,27 @@
+        :key="change.name"
+        :change="change"
+      >
+      </ChangeItem> -->
```
Do not leave commented out code.

rbclark commented on src/components/cards/comparison/ProfileRow.vue:
```diff
@@ -0,0 +1,27 @@
+<template>
+  <v-col cols="4" xs="4" sm="3" md="3" lg="3" xl="3">
```
See previous comment about v-cols.

rbclark commented on src/components/cards/controltable/ControlRowCol.vue:
```diff
@@ -9,8 +9,8 @@
+      xl="10"
       cols="12"
       sm="12"
       md="12"
```
should be replaced with `cols="12" lg="10"`.

rbclark commented on src/components/cards/controltable/ControlRowCol.vue:
```diff
@@ -35,7 +35,7 @@
       </v-clamp>
     </v-col>
-    <v-col v-else lg="6" xl="6" sm="6" md="6" cols="12" class="left">
+    <v-col v-else lg="5" xl="5" sm="6" md="6" cols="12" class="left">
```
This can be replaced with `<v-col v-else cols="12" sm="6" lg="5"  class="left">`

Although I'm not sure I understand the `lg="5"` since that will leave a blank 2 columns on large and xl screens I believe.-

rbclark commented on src/components/cards/controltable/ControlRowDetails.vue:
```diff
@@ -222,8 +242,12 @@ export default class ControlRowDetails extends ControlRowDetailsProps {
+  //for zebra background
   zebra(ix: number): string {
-    return ix % 2 ? '' : 'zebra-table';
+    if (ix % 2 == 0) {
+      return 'zebra-table';
+    }
+    return 'non-zebra-table';
```
Why was the ternary operator replaced here? `return ((ix % 2 == 0) ? 'non-zebra-table' : 'zebra-table');` should be equivalent.

rbclark commented on src/components/generic/ApexLineChart.vue:
```diff
@@ -0,0 +1,219 @@
+
+  //changes text based on light or dark mode
+  get white_black(): string {
+    var whiteBlack;
```
Is this variable used anywhere?

cjdoherty replied:
yes, line 172, 197, 204, 209

rbclark replied:
That uses the `white_black` method however I don't see anything getting or setting the `whiteBlack` variable.

cjdoherty committed:
- added sort by control in compare
- shows only 2 controls
- fixed resizing
- added metadata change
- made just one comparison select view
- removed status from metadata change

samuelsunvold committed:
Inital unit test for Modal.vue. Not working correctly/still in progress

Signed-off-by: Sam Sunvold <sunvoldsj@vcu.edu>

samuelsunvold committed:
Update Modal.spec.ts

cjdoherty committed:
made all tests pass so netlify will build

cjdoherty committed:
merge con

cjdoherty committed:
- compare tests (more tests on the way)
- utilities and cleanup

cjdoherty committed:
most of the tests for results.vue components

cjdoherty committed:
datatable tests

cjdoherty committed:
some clean up and a couple more tests

cjdoherty committed:
test cleanup

cjdoherty committed:
Merge branch 'master' of https://github.com/mitre/heimdall-lite into delta-view

cjdoherty committed:
Merge branch 'master' of https://github.com/mitre/heimdall-lite into delta-view

cjdoherty committed:
updated counting in status chart

cjdoherty committed:
Merge branch 'delta-view' of https://github.com/mitre/heimdall-lite into delta-view

cjdoherty committed:
updated counting in tests and compare

Signed-off-by: Colin Doherty <oneofthree103@gmail.com>

cjdoherty committed:
cleaned up unused code

Signed-off-by: Colin Doherty <oneofthree103@gmail.com>

cjdoherty committed:
cleanup

Signed-off-by: Colin Doherty <oneofthree103@gmail.com>

cjdoherty committed:
Merge branch 'master' into delta-view

cjdoherty committed:
more cleanup

Signed-off-by: Colin Doherty <oneofthree103@gmail.com>

cjdoherty committed:
Merge branch 'delta-view' of https://github.com/mitre/heimdall-lite into delta-view

cjdoherty committed:
more cleanup

Signed-off-by: Colin Doherty <oneofthree103@gmail.com>

cjdoherty committed:
uninstall unused npm packages

Signed-off-by: Colin Doherty <oneofthree103@gmail.com>

rbclark commented on src/components/generic/ApexLineChart.vue:
```diff
@@ -0,0 +1,218 @@
+
+    // If we have any non-zero data, just returngive 0.01 of all
+    return final;
+  }
```
Can we take advantage of [vue-property-validation](https://www.digitalocean.com/community/tutorials/vuejs-property-validation) here? Also, if we mark these fields as required then there should never be a case where it isn't an instance of array due to the property declaration.

cjdoherty replied:
Right now moving validation to props is throwing ts compiler errors, I think this might be better to fix when we switch to using vue-property-decorator.

rbclark commented on src/components/generic/ApexLineChart.vue:
```diff
@@ -0,0 +1,218 @@
+
+  get label_colors(): string[] {
+    let colors = [];
+    for (let i = 0; i < this._categories.length; i++) {
```
Using Vue validation will allow this to just be `this.categories.length`.

cjdoherty replied:
vue-class-component does not give type checking, it assumes category is an array of any, so this is better solved when we implement the vue-property-decorator, many other components outside or inside my branch have this issue, so I think it might be better to fix it all at once in a different pr

rbclark commented on src/components/generic/ApexLineChart.vue:
```diff
@@ -0,0 +1,218 @@
+    colors.push('#FF9800');
+    colors.push('#FF5722');
+    colors.push('#F44336');
+    return colors;
```
Shouldn't need to create an empty array and push here, should be able to just

```javascript
return new Array('#FFEB3B', '#FF9800', '#FF5722', '#F44336');
```

rbclark commented on src/components/generic/ApexLineChart.vue:
```diff
@@ -0,0 +1,218 @@
+  }
+
+  get line_colors(): string[] | undefined {
+    let colors = [];
```
Unnecessary variable here.

rbclark commented on src/components/global/ExportJson.vue:
```diff
@@ -36,40 +39,62 @@ const Props = Vue.extend({
+          saveAs(new Blob([b]), 'exported_jsons.zip');
+        })
+      );
+      zipfile.end();
```
Having the same 2 loops copy pasted here twice seems difficult to update down the road. It would be useful to instead count the number of elements after the fact instead of duplicating all of the loop logic twice.

```javascript
      let fileData = new Array<FileData>();
      for (let evaluation of filter_mod.evaluations(ids)) {
        fileData.push({name: evaluation.from_file.filename, contents: JSON.stringify(evaluation.data)})
      }
      for (let prof of filter_mod.profiles(ids)) {
        if (isFromProfileFile(prof)) {
          fileData.push({name: prof.from_file.filename, contents: JSON.stringify(prof.data)})
        }
      }
```

The type declaration for FileData is

```javascript
type FileData = {
  name: string
  contents: string
}
```

From there the if else logic should be similar, just inspect the length of `fileData` instead of the number of IDs.

rbclark commented on src/components/global/upload_tabs/SampleList.vue:
```diff
@@ -49,47 +65,65 @@ const Props = Vue.extend({
       {
-        name: 'RedHat 7 STIG Baseline InSpec Sample',
-        url: `https://raw.githubusercontent.com/${this.repo}/master/samples/rhel7-results.json`
+        name: 'RedHat 7 STIG Baseline',
```
RedHat should have a space in it, this looks like the only one without it.

rbclark commented on src/components/global/Sidebar.vue:
```diff
@@ -72,6 +96,33 @@ const SidebarProps = Vue.extend({
+    ) {
+      return true;
+    }
+    return false;
```
`return data_module.allFiles.length == filtered_module.selected_file_ids.length` is equivalent to this.

rbclark commented on src/components/global/upload_tabs/SampleList.vue:
```diff
@@ -99,35 +133,48 @@ export default class SampleList extends Props {
     return `${mod.repo_org}/${mod.repo_name}`;
   }
```
I don't believe this function is used anywhere anymore since all the files are pulled locally.

rbclark commented on src/components/global/upload_tabs/SampleList.vue:
```diff
@@ -99,35 +133,48 @@ export default class SampleList extends Props {
+      }
+    }
+    this.selected_samps.push(samp);
+  }
```
This is a bit more readable using array `indexOf`:

```javascript
    let checked = this.selected_samps.indexOf(samp);
    // If the sample is currently checked then this is a toggle to uncheck it
    // indexOf returns -1 if the element is not in the list
    if (checked != -1) {
      this.selected_samps.splice(checked);
    }
    else // Otherwise this is a toggle to check it, added to the selected samples list
    {
      this.selected_samps.push(samp);
    }
```

rbclark commented on src/store/data_filters.ts:
```diff
@@ -83,47 +83,75 @@ function contains_term(
+      if (this.selected_file_ids[i] == file) {
+        this.selected_file_ids.splice(i, 1);
+      }
+    }
```
Would be better as an `indexOf` same as above. Also the default for splice is `1` element.

rbclark commented on src/utilities/delta_util.ts:
```diff
@@ -184,44 +170,44 @@ export class ControlDelta {
+    for (let ctrl of prof.contains) {
+      if (ctrl.hdf.segments!.length) {
+        let t = ctrl.hdf.segments![0].start_time;
+        return t;
```
No need for this to be 2 lines, just return the time. `return ctrl.hdf.segments![0].start_time;`

rbclark commented on src/views/Results.vue:
```diff
@@ -366,15 +393,54 @@ export default class Results extends ResultsProps {
-      this.$router.push(`/results/${ids[0]}`);
+    //enable all uploaded files
+    let filter_module = getModule(FilteredDataModule, this.$store);
+    for (let i of ids) {
+      filter_module.set_toggle_file_on(i);
+    }
```
There's 2 spots in the code with this exact same loop, it might be worthwhile to just have the function take an array of ids you want to enable.

rbclark commented on tests/util/testingUtils.ts:
```diff
@@ -0,0 +1,156 @@
+    unique_id: next_free_file_ID(),
+    text: JSON.stringify(sample.sample)
+  });
+}
```
Is it possible to reuse any of the declarations in `SampleList.vue` here so we don't have all of the samples and loading logic duplicated?

cjdoherty committed:
resolved Robs review, mostly cleanup

Signed-off-by: Colin Doherty <oneofthree103@gmail.com>

cjdoherty committed:
clean up in ApexLineChart

Signed-off-by: Colin Doherty <oneofthree103@gmail.com>

cjdoherty committed:
fixed tests errors

Signed-off-by: Colin Doherty <oneofthree103@gmail.com>

cjdoherty committed:
more ApexLineChart cleanup

Signed-off-by: Colin Doherty <oneofthree103@gmail.com>

rbclark commented:
Looks like https://github.com/mitre/heimdall-lite/pull/294 was reverted as part of this PR.

aaronlippold reviewed (dismissed):
@rbclark @cjdoherty @Mitriol Approve as long as any outstanding issues are agreed to you three with a planned path to resolution and issues to track them.

Commit:
Make test status message unclickable

This was done in PR #294 but got reverted in this PR

Signed-off-by: Robert Clark <rbclark@mitre.org>
